waiter/src/waiter/service.clj

```diff
@@ -226,39 +225,37 @@
-                        (do
-                          ;; No instances were started in a reasonable amount of time
-                          (meters/mark! (metrics/service-meter service-id "no-available-instance-timeout"))
-                          (statsd/inc! metric-group "no_instance_timeout")
-                          (let [outstanding-requests (counters/value (metrics/service-counter service-id "request-counts" "outstanding"))
-                                requests-waiting-to-stream (counters/value (metrics/service-counter service-id "request-counts" "waiting-to-stream"))
-                                waiting-for-available-instance (counters/value (metrics/service-counter service-id "request-counts" "waiting-for-available-instance"))
-                                unhealthy-instances (counters/value (metrics/service-counter service-id "instance-counts" "unhealthy"))]
-                            (ex-info (str "After " (t/in-seconds (t/millis queue-timeout-ms))
-                                          " seconds, no instance available to handle request."
-                                          (when (and (zero? healthy-instances) (or (pos? unhealthy-instances) (pos? failed-instances)))
-                                            " Check that your service is able to start properly!")
-                                          (when (and (pos? outstanding-requests) (pos? healthy-instances))
-                                            " Check that your service is able to scale properly!"))
-                                     {:service-id service-id
-                                      :outstanding-requests outstanding-requests
-                                      :requests-waiting-to-stream requests-waiting-to-stream
-                                      :waiting-for-available-instance waiting-for-available-instance
-                                      :slots-assigned (counters/value (metrics/service-counter service-id "instance-counts" "slots-assigned"))
-                                      :slots-available (counters/value (metrics/service-counter service-id "instance-counts" "slots-available"))
-                                      :slots-in-use (counters/value (metrics/service-counter service-id "instance-counts" "slots-in-use"))
-                                      :work-stealing-offers-received (counters/value (metrics/service-counter service-id "work-stealing" "received-from" "in-flight"))
-                                      :work-stealing-offers-sent (counters/value (metrics/service-counter service-id "work-stealing" "sent-to" "in-flight"))
-                                      :status 503})))
-                        (do
-                          (app-not-found-fn)
-                          (async/<! (async/timeout 1500))
-                          (recur (inc iterations)))))))))))
+                    (if-not (t/before? (t/now) expiry-time) 
```

Comment:
Should simplify to use `(if (t/after...))`?

Reply:
That would not be true when they are the same.  It's subtle but the test should fail if they are equal so I'd rather leave it.